app/Http/Actions/BuildAchievementOfTheWeekDataAction.php

```diff
@@ -24,25 +21,8 @@ public function execute(?StaticData $staticData): ?EventAchievementData
             ->with(['achievement.game', 'sourceAchievement.game'])
             ->first();
 
-        if (!$achievementOfTheWeek || !$achievementOfTheWeek->source_achievement_id) {
```

Comment:
This still needs to ensure there's a source achievement, or an exception occurs on the main page.

Reply:
Resolved in latest. I've also added another test case to `HomeControllerTest` to ensure this doesn't accidentally get broken again.